data/examples/declaration/signature/fixity/infix.hs

```diff
@@ -1,2 +1,4 @@
 infix 0 <?>
 infix 9 <^-^>
+
+infix 2 ->
```

Comment:
Was it previously impossible to specify a custom fixity for `->`?

Reply:
Yes and no: `infix 2 ->` could be parsed also by older GHCs, but all GHCs later reject it ("`Illegal fixity signature of built-in syntax: ->`"). I only included this as I had to add this new case in `p_rdrName`
```haskell
          -- whether the `->` identifier is parenthesized
          NameAnnRArrow {nann_mopen = Just _} -> parens N
```
and I wondered when it is possible for `->` (as an identifier, so not as part of a normal function arrow `A -> B`, which is parsed to `HsFunTy`) to *not* be parenthesized, and a fixity declaration was the only thing I could came up with, so I added a test for it right here.